cloud/ec2_ami_copy.py

```diff
@@ -0,0 +1,224 @@
+    required: false
+    default: null
+    aliases: []
+
+author: Amir Moulavi <amir.moulavi@gmail.com>
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Basic AMI Copy
+- local_action:
+    module: ec2_ami_copy
+    aws_access_key: "-----REMOVED-------"
+    aws_secret_key: "-----REMOVED-------"
+    source_region: eu-west-1
+    dest_region: us-east-1
+    source_image_id: ami-xxxxxxx
+    name: SuperService-new-AMI
+    description: latest patch
+    tags: '{"Name":"SuperService-new-AMI", "type":"SuperService"}'
+    wait: yes
+  register: image_id
+'''
+
+
+import sys
+import time
+
+try:
+    import boto
+    import boto.ec2
+except ImportError:
+    print "failed=True msg='boto required for this module'"
+    sys.exit(1)
+
+
+def copy_image(module, ec2):
+    """
+    Copies an AMI
+
+    module : AnsibleModule object
+    ec2: authenticated ec2 connection object
+    """
+
+    source_region = module.params.get('source_region')
+    source_image_id = module.params.get('source_image_id')
+    name = module.params.get('name')
+    description = module.params.get('description')
+    tags = module.params.get('tags')
+    wait_timeout = int(module.params.get('wait_timeout'))
+    wait = module.params.get('wait')
+
+    try:
+        params = {'source_region': source_region,
+                  'source_image_id': source_image_id,
+                  'name': name,
+                  'description': description
+        }
+
+        image_id = ec2.copy_image(**params).image_id
+    except boto.exception.BotoServerError, e:
+        module.fail_json(msg="%s: %s" % (e.error_code, e.error_message))
+
+    img = wait_until_image_is_recognized(module, ec2, wait_timeout, image_id, wait)
+
+    img = wait_until_image_is_copied(module, ec2, wait_timeout, img, image_id, wait)
+
+    register_tags_if_any(module, ec2, tags, image_id)
+
+    module.exit_json(msg="AMI copy operation complete", image_id=image_id, state=img.state, changed=True)
+
+
+# register tags to the copied AMI in dest_region
+def register_tags_if_any(module, ec2, tags, image_id):
+    if tags:
+        try:
+            ec2.create_tags([image_id], tags)
+        except Exception as e:
+            module.fail_json(msg=str(e))
+
+
+# wait here until the image is copied (i.e. the state becomes available
+def wait_until_image_is_copied(module, ec2, wait_timeout, img, image_id, wait):
+    wait_timeout = time.time() + wait_timeout
+    while wait and wait_timeout > time.time() and (img is None or img.state != 'available'):
+        img = ec2.get_image(image_id)
+        time.sleep(3)
+    if wait and wait_timeout <= time.time():
+        # waiting took too long
+        module.fail_json(msg="timed out waiting for image to be copied")
+    return img
+
+
+# wait until the image is recognized.
+def wait_until_image_is_recognized(module, ec2, wait_timeout, image_id, wait):
+    for i in range(wait_timeout):
```

Comment:
No, we need to wait for image to be recognized by Amazon so that we can return correct info about the image. Wait is only used when we need to wait for the copy operation to be finished